app/models/progress_message.rb

```diff
@@ -5,6 +5,8 @@ class ProgressMessage < ApplicationRecord
 
   AS_JSON_ATTRIBUTES = %w(id level message received_at).freeze
 
+  scope :by_owner, -> { joins(:order_item).select("#{table_name}.*, order_items.owner").where('owner = ?', ManageIQ::API::Common::Request.current.user.username) }
+
```

Comment:
Would be nice to see specs around this, only to prove the SQL works.

Reply:
Added some speccy specs